frontend2/src/components/team/EligibilitySettings.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from "react";
+import {
+  useCurrentTeam,
+  TeamStateEnum,
+} from "../../contexts/CurrentTeamContext";
+import SectionCard from "../SectionCard";
+import DescriptiveCheckbox, {
+  CheckboxState,
+} from "../elements/DescriptiveCheckbox";
+import { useEpisode, useEpisodeId } from "../../contexts/EpisodeContext";
+import { isPresent } from "../../utils/utilTypes";
+import { updateTeamPartial } from "../../utils/api/team";
+
+export function determineCheckboxState(
+  inDesired: boolean,
+  inActual: boolean,
+): CheckboxState {
+  if (inDesired && inActual) {
+    return CheckboxState.CHECKED;
+  } else if (!inDesired && !inActual) {
+    return CheckboxState.UNCHECKED;
+  } else {
+    return CheckboxState.LOADING;
+  }
+}
+
+// This component should only be used when there is a logged in user with a team.
+const EligibilitySettings: React.FC = () => {
+  const { team, teamState, refreshTeam } = useCurrentTeam();
+  const episode = useEpisode();
+  const { episodeId } = useEpisodeId();
+  // the desired state according to what the user has clicked on the page
+  const [desiredEligibility, setDesiredEligibility] = useState<
+    number[] | undefined
+  >();
+  useEffect(() => {
+    setDesiredEligibility(team?.profile?.eligible_for);
+  }, [team?.profile?.eligible_for]);
+
+  useEffect(() => {
+    // if actual team eligibility hasn't loaded yet or the desired/actual are identical
+    if (
+      (!isPresent(desiredEligibility) ||
+        desiredEligibility === team?.profile?.eligible_for) ??
+      []
+    ) {
+      return;
+    }
+
+    let isActive = true;
+    const update = async (): Promise<void> => {
+      const updatedTeam = await updateTeamPartial(episodeId, {
+        eligible_for: desiredEligibility,
+      });
+      const newProfile = team?.profile;
+      if (isActive && isPresent(newProfile) && isPresent(team)) {
+        // only update the eligibility of the team profile to avoid race conditions
+        // with other team profile updaters on this page
+        newProfile.eligible_for = updatedTeam.profile?.eligible_for;
+        refreshTeam({
+          ...team,
+          profile: newProfile,
+        });
+      }
+    };
+    void update();
+    return () => {
+      isActive = false;
+    };
+  }, [desiredEligibility, episodeId]);
+
+  if (
+    teamState !== TeamStateEnum.IN_TEAM ||
+    !isPresent(team) ||
+    !isPresent(episode)
+  ) {
+    return null;
```

Comment:
Should we have a message here to say "you're not in a team!" or smth?

Reply:
I can return a string that says that. 
I also have a comment on the component that says the current user should be in a team